src/coreclr/nativeaot/Runtime/arm64/WriteBarriers.asm

```diff
@@ -362,8 +363,6 @@ ExchangeRetry
 
         INSERT_CHECKED_WRITE_BARRIER_CORE x14, x15, x15
 
-        add     x14, x14, #8
```

Comment:
`JIT_ByRefWriteBarrier` (= `RhpByRefAssignRefArm64`) is not using this function, so this increment is not needed.